### ISSUE TITLE
Don't exit if a corrupt index is found

### ISSUE THREAD

**Issue by garethbowles** [etsy/Hound#70]
I'm setting up Hound to run against about 3000 Stash repos (PR coming to handle the different file link format) and noticed that a small number of repos weren't indexed properly the first time.  As soon as Hound finds a bad index, it immediately shuts down.

Deleting the source code and index for each repo and letting Hound re-clone and reindex seemed to work every time.

I'd prefer not to have Hound exit as soon as it finds a bad index, but log a severe error instead. Rather than submitting a PR for this behaviour, I wanted to check what other people think first - maybe the option should be configurable for example ?

**Comment by kellegous** [comment]
My opinion on what hound should do:

 1. When a corrupted index is detected, hound should start from scratch on the reindex and try to recover gracefully, though slowly.
 2. Waiting on all this stuff at startup really kind of sucks, but was just a stop-gap to make sure the server came up in a decent state (i.e. all searchers had a searchable index). The next logical step is to bring the server up when the searchers were still building their indices and have searcher state indicating whether they are offline or not.
 3. The biggest issue associated with all of this stuff is properly communicating in the UI that a repository is temporarily offline. I don't want users searching thinking they are covering all repos when, in fact, some are not searchable at all.

**Comment by garethbowles** [comment]
Good point on communicating that a repo is offline.  I think it's OK to not bring the server up until the initial index is done for all repos, but if there is an indexing error Hound should definitely communicate somehow that results from the affected repo aren't available.  This would be a big improvement on OpenGrok (which we're hoping to replace with Hound at Netflix), where indexing errors are completely opaque to the UI.

**Comment by aj-jester** [comment]
Would love to see this. Been experiencing this same issue but with a lot less repos, < 100.

**Comment by Wilfred** [comment]
> Waiting on all this stuff at startup really kind of sucks, but was just a stop-gap to make sure the server came up in a decent state (i.e. all searchers had a searchable index).

I have an instance with ~600 repos and this means restarting the service is very slow (can take more than 10 minutes). Should I open a separate issue for this?
